src/routes.rs

```diff
@@ -26,6 +26,22 @@ pub fn create_router(state: Arc<AppState>) -> Router {
     }
 
     let pipeline_router = Steer::new(routers, move |req: &Request, _services: &[_]| {
+        let path = req.uri().path();
+
+        // First try routing by path
+        if path.contains("/chat/completions") {
+            return *pipeline_idxs.get("default").unwrap_or(&0);
+        } else if path.contains("/audio") {
+            return *pipeline_idxs.get("audio").unwrap_or(&0);
+        } else if path.contains("/embeddings") {
+            return *pipeline_idxs.get("embeddings").unwrap_or(&0);
+        } else if path.contains("/image") {
+            return *pipeline_idxs.get("image").unwrap_or(&0);
+        } else if path.contains("/video") {
+            return *pipeline_idxs.get("video").unwrap_or(&0);
+        }
+
+        // Fall back to header-based routing if needed
```

Comment:
this (suspiciously cursor generated) code change the routing behavior of the whole app....